docs/src/description/declarations.md

```diff
@@ -1,22 +1,36 @@
 # Type declaration sections
 
-## Trace columns (`trace_columns`)
+## Constants (`const`)
+
+Constants can be optionally declared with the const keyword at the top of an AirScript module just below the declaration of the module name. They can be scalars, vectors or matrices. Constant names must contain only uppercase letters.
+
+Each constant is defined by an identifier and a value in the following format:
+
+```
+const FOO = 123
+const BAR = [1, 2, 3]
+const BAZ = [[1, 2, 3], [4, 5, 6]]
+```
+
+In the above example, `FOO` is a constant of type scalar with value `123`, BAR is a constant of type vector with value `[1, 2, 3]`, and BAZ is a constant of type matrix with value `[[1, 2, 3], [4, 5, 6]]`.
+
+## Execution trace (`trace_columns`)
 
 A `trace_columns` section contains declarations for `main` trace columns or `aux` (auxiliary) trace columns.
 
-The `main` and `aux` declarations define the shape of the main and auxiliary execution traces respectively and declare identifiers which can be used to refer to each of the columns in that trace.
+The `main` and `aux` declarations define the shape of the main and auxiliary execution traces respectively and define identifiers which can be used to refer to each of the columns or a group of columns in that trace. The columns can also be referred using built-in variables `$main` and `$aux` and the index of the column in the respective trace.
```

Comment:
nit: instead of "...referred using built-in variables `$main`..." this should be "...referenced using the built-in variables `$main`..."